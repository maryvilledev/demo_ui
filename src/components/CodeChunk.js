import React from 'react'

const CodeChunk = ({concept, lang, color, children, onMouseEnter}) => (
  <span
    style={{backgroundColor: color}}
    onMouseEnter={() => onMouseEnter(lang, concept)}
    >{children}</span>
)

CodeChunk.propTypes = {
  color: React.PropTypes.string,
  concept: React.PropTypes.string,
  lang: React.PropTypes.string,
  children: React.PropTypes.any,
  onMouseEnter: React.PropTypes.func,
}

export default CodeChunk;
