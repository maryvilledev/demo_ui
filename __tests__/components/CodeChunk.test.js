import React from 'react'
import ReactDOM from 'react-dom'
import CodeChunk from '../../src/components/CodeChunk'

import testRenderer from 'react-test-renderer'

const mockConcept = 'fat-arrow'
const mockLang = 'javascript'
const mockColor= '#FFFFFF'
const mockOnMouseEnter = (concept, lang) => `${concept}-${lang}`;

describe('<CodeChunk />', () => {
  it('renders correctly', () => {
    const tree = testRenderer.create(
      <CodeChunk
      concept={mockConcept}
      lang={mockLang}
      color={mockColor}
      onMouseEnter={mockOnMouseEnter} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
