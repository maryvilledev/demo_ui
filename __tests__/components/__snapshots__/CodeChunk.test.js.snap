exports[`<CodeChunk /> renders correctly 1`] = `
<span
  onMouseEnter={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  } />
`;
